client/components/checkbox/index.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+* External dependencies
+*/
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+/**
+* Interal dependencies
+*/
+import Gridicon from 'components/gridicon';
+
+export default React.createClass( {
+	displayName: 'Checkbox',
+
+	propTypes: {
+		checked: PropTypes.bool.isRequired,
+		disabled: PropTypes.bool,
+		onChange: PropTypes.func
+	},
+
+	getInitialState() {
+		return {
+			focused: false
+		}
+	},
+
+	getDefaultProps() {
+		return {
+			disabled: false
+		};
+	},
+
+	handleChange( e ) {
+		const { onChange } = this.props;
+
+		onChange && onChange( e, this.state.checked );
+	},
+
+	handleFocus() {
+		this.setState( {
+			focused: true
+		} )
+	},
+
+	handleBlur() {
+		this.setState( {
+			focused: false
+		} )
+	},
+
+	render() {
+		const { focused } = this.state;
+		const { className, children, checked, disabled, id, name } = this.props;
```

Comment:
Oops - don't forget to put these (`id` and `name`) in the `propTypes` declaration and documentation! 😉